### ISSUE TITLE
Remove old files from S3 subfolder before upload new ones

### ISSUE THREAD

**Issue by make-ing** [Jefiozie/ngx-aws-deploy#104]
**Is your feature request related to a problem? Please describe.**
I am using ngs-aws-deploy now for some time and my S3 folder is starting to get filled with old build files from previous versions of my SPA. Because Angular is creating the files with some sort of random indetifier to prevent caching, the "old" files dont get deleted when i deploy the new version.

**Describe the solution you'd like**
The deployment script should delete all files currently in the subFolder and the upload the new ones.

**Describe alternatives you've considered**
Add an option to enable the deleting of the old files before the upload of the new ones.

**Additional context**
N/A

**Comment by Jefiozie** [comment]
Hi @make-ing, I'm not completely sure if we should support this feature. I see some problems when you have a bucket with multiple folders and accidentally you switch the flag (but maybe that is just me). However, did you know that you can use ` ng build --output-hashing none` this will allow you to always have the same names and not the unique names that are generated by the Angular CLI. Maybe this is a quick workaround for you? Let me know what you think.

![image](https://user-images.githubusercontent.com/17835373/80565046-cc89e680-89ef-11ea-8550-961e8f4f418e.png)


I will add a discussion tag on this issue so that people can comment if we should add this to the library.

**Comment by make-ing** [comment]
First of all. Thank you for your response. It is not normal nowadays to find a good and well maintained project. Or its just me finding not the right ones.

I thought of removing the hasing as well, but its actually very usefull to counter the problem of caching and stuff. For now its not a big Problem, but i realy would like to have such a feature. Maybe you can add a confirmation input when the flag is used?

**Comment by brianfromoregon** [comment]
I would use the flag if you added it. Thanks for this great project btw!

**Comment by Jefiozie** [comment]
I'm adding a tag for `help wanted` is one of you interested in making a PR for this?

**Comment by make-ing** [comment]
At the moment i don't have time for this, which would take me quite a while because i'm more of a backend guy.

**Comment by Jefiozie** [comment]
After #173 I think somebody could pickup this issue

**Comment by rlmestre** [comment]
When using AWS CLI S3 Sync, or with the DataSync service (not applicable here), there are flags that mimic this exact behavior: --delete removes all files not in the locally pushed directory. 

One alternative in this case would be a config item with a couple options. You can probably get ListObjects and diff it with your array of local files and delete everything not listed. Another option would be to just delete everything before uploading.

I might be able to submit a pull request soon. But no promises 😄

**Comment by Jefiozie** [comment]
Hi @rlmestre, I would love to see this landed in a PR. If we could adopt it from the `aws-cli`  and have a way to do this from our own source when `aws-cli` is not installed it would be great.

**Comment by aviboy2006** [comment]
I would like to pick this. i have approach like we can move existing file to backup_date folder so this can help to revert if we deployment failed or code failed to roll back.

**Comment by Jefiozie** [comment]
Interesting approach, looking forward on reviewing your PR.

**Comment by thomasturrell** [comment]
I believe that the assets folder should be an optional exception. The project that I am working on uses the assets folder to load the environment configuration from. The environmental configuration is not part of the build. I believe that this is quite a common approach.

I believe that only the files that were added in the previous deployment should be deleted by the next deployment.

**Comment by mcandre** [comment]
In fact, it is preferable to remove stale files *after* uploading the new files. This way avoids many availability problems.

**Comment by phhien203** [comment]
May I know the current status of this discussion @Jefiozie ?

**Comment by rlmestre** [comment]
I opened a draft PR for this (WIP, untested) with options for deleting before or after. @Jefiozie let me know your thoughts.
